pkg/controller/v1alpha2/applicationconfiguration/applicationconfiguration.go

```diff
@@ -247,13 +255,26 @@ type Workload struct {
 	// A Workload object.
 	Workload *unstructured.Unstructured
 
+	// Unready indicates whether this resource is unready to apply
+	// because of dependency not ready
+	Unready bool
+
 	// Traits associated with this workload.
-	Traits []unstructured.Unstructured
+	Traits []*Trait
 
 	// Scopes associated with this workload.
 	Scopes []unstructured.Unstructured
 }
 
+// A Trait produced by an OAM ApplicationConfiguration.
+type Trait struct {
+	Object unstructured.Unstructured
+
+	// Unready indicates whether this resource is unready to apply
+	// because of dependency not ready
+	Unready bool
```

Comment:
What if the outside Unready is true while the `Unready` field here is false? Does it mean workload will apply before all trait is ready?

Reply:
This does not have any indication on ordering. It will comply with whatever ordering being applied but skip unready ones.